src/translator/cache.cpp

```diff
@@ -0,0 +1,169 @@
+#include "cache.h"
+
+#include <cstdlib>
+
+#include "translation_model.h"
+
+namespace marian {
+namespace bergamot {
+
+size_t HashCacheKey::operator()(const CacheKey &key) const {
```

Comment:
Does this work with -m32 option? `size_t` is `uint32_t` there, whereas we expect `uint64_t` keys.
I added `m32` build option to marian: https://github.com/browsermt/marian-dev/blob/master/CMakeLists.txt#L40

Reply:
Everything's made `size_t`. 

[`hash_combine`](https://github.com/marian-nmt/marian-dev/blob/9fa166be885b025711f27b35453e0f2c00c9933e/src/common/hash.h) mixes up `size_t` seeds and such unless explicitly specified. Inconsistency would lead to issues when going between WASM and Native. 

The rationale applied is to follow `size_t` everywhere (even cpp `std::hash` returns `size_t`). For WASM this will consistently be 32-bit while for Native it'll be 64-bit, which should avoid any narrowing errors and inconsistencies.